src/runtime_src/core/tools/xbmgmt2/OO_Hotplug.cpp

```diff
@@ -0,0 +1,162 @@
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "OO_Hotplug.h"
+#include "tools/common/XBUtilitiesCore.h"
+#include "tools/common/XBUtilities.h"
+
+#include "core/common/system.h"
+#include "core/common/query_requests.h"
+
+// 3rd Party Library - Include Files
+#include <boost/program_options.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+namespace po = boost::program_options;
+
+// =============================================================================
+
+// ----- C L A S S   M E T H O D S -------------------------------------------
+
+OO_Hotplug::OO_Hotplug( const std::string &_longName, bool _isHidden )
+    : OptionOptions(_longName, _isHidden, "Perform hotplug for the given device")
+{
+  m_optionsDescription.add_options()
+    ("device,d", boost::program_options::value<decltype(m_devices)>(&m_devices)->multitoken(), "The Bus:Device.Function (e.g., 0000:d8:00.0) device of interest")
+    ("action", boost::program_options::value<decltype(m_action)>(&m_action)->required(), "Action to perform: online or offline")
+    ("help", boost::program_options::bool_switch(&m_help), "Help to use this sub-command")
+  ;
+
+  m_positionalOptions.
+    add("action", 1 /* max_count */)
+  ;
+}
+
+void
+OO_Hotplug::execute(const SubCmdOptions& _options) const
+{
+  XBUtilities::verbose("SubCommand option: Hotplug");
+
+  XBUtilities::verbose("Option(s):");
+  for (auto & aString : _options)
+    XBUtilities::verbose(std::string(" ") + aString);
+
+  // Honor help option first
+  if (std::find(_options.begin(), _options.end(), "--help") != _options.end()) {
+    printHelp();
+    return;
+  }
+
+  // Parse sub-command ...
+  po::variables_map vm;
+
+  try {
+    po::store(po::command_line_parser(_options).options(m_optionsDescription).positional(m_positionalOptions).run(), vm);
+    po::notify(vm); // Can throw
+  }
+  catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << "\n\n";
+    printHelp();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  if(m_help) {
+    printHelp();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+  // Exit if neither action or device specified
+  if(m_action.empty()) {
+    printHelp();
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+
+  try {
+    if (!boost::iequals(m_action, "online") && !boost::iequals(m_action, "offline")) {
+      std::cerr << boost::format("ERROR: Invalid action value: '%s'\n") % m_action;
+      printHelp();
+      throw xrt_core::error(std::errc::operation_canceled);
+    }
+    bool is_online = boost::iequals(m_action, "online");
```

Comment:
const and referece? it is better to check within 98 line. No need to create this extra variable